src/test/java/org/opensearch/security/auditlog/compliance/ComplianceConfigTest.java

```diff
@@ -136,4 +152,84 @@ public void testEmpty() {
         assertSame(WildcardMatcher.NONE, complianceConfig.getIgnoredComplianceUsersForWriteMatcher());
     }
+
+    @Test
+    public void testLogState() {
+        // arrange
+        final var logger = Mockito.mock(Logger.class);
+        final ComplianceConfig complianceConfig = ComplianceConfig.from(Settings.EMPTY);
+        // act
+        complianceConfig.log(logger);
+        // assert: don't validate content, but ensure message's logged is generally consistant
+        verify(logger, times(6)).info(anyString(), anyString());
+        verify(logger, times(1)).info(anyString(), isNull(String.class));
+        verify(logger, times(1)).info(anyString(), any(Map.class));
+        verify(logger, times(3)).info(anyString(), any(WildcardMatcher.class));
+        verifyNoMoreInteractions(logger);
+    }
+
+    @Test
+    public void testReadWriteHistoryEnabledForIndex_rollingIndex() {
+        // arrange
+        final var date = new AtomicReference<DateTime>();
+        final Consumer<Integer> setYear = (year) -> { date.set(new DateTime(year, 1, 1, 1, 2, DateTimeZone.UTC)); };
+        final ComplianceConfig complianceConfig = ComplianceConfig.from(
+            Settings.builder()
+                .put(
+                    ConfigConstants.SECURITY_AUDIT_CONFIG_DEFAULT_PREFIX + ConfigConstants.SECURITY_AUDIT_OPENSEARCH_INDEX,
+                    "'audit-log-index'-YYYY-MM-dd"
+                )
+                .put(ConfigConstants.SECURITY_AUDIT_TYPE_DEFAULT, "internal_opensearch")
+                .putList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_READ_WATCHED_FIELDS, "*")
+                .putList(ConfigConstants.OPENDISTRO_SECURITY_COMPLIANCE_HISTORY_WRITE_WATCHED_INDICES, "*")
+                .build(),
+            date::get
+        );
+
+        // act: Don't log for null indices
+        assertThat(complianceConfig.readHistoryEnabledForIndex(null), equalTo(false));
+        assertThat(complianceConfig.writeHistoryEnabledForIndex(null), equalTo(false));
+        // act: Don't log for the security indices
+        assertThat(complianceConfig.readHistoryEnabledForIndex(complianceConfig.getSecurityIndex()), equalTo(false));
+        assertThat(complianceConfig.writeHistoryEnabledForIndex(complianceConfig.getSecurityIndex()), equalTo(false));
+
+        // act: Don't log for the current audit log
+        setYear.accept(1337);
+        assertThat(complianceConfig.readHistoryEnabledForIndex("audit-log-index-1337-01-01"), equalTo(false));
+        assertThat(complianceConfig.writeHistoryEnabledForIndex("audit-log-index-1337-01-01"), equalTo(false));
+
+        // act: Log for current audit log when it does not match the date
+        setYear.accept(2048);
+        // See https://github.com/opensearch-project/security/issues/3950
+        // assertThat(complianceConfig.readHistoryEnabledForIndex("audit-log-index-1337-01-01"), equalTo(true));
```

Comment:
Writing this unit test exposed a bug in the implementation around roll over logs :( I've filed an issue [1] and we can follow up separately.

- [1] https://github.com/opensearch-project/security/issues/3950